.env.example

```diff
@@ -13,12 +13,12 @@ VALIDATOR_ADDRESS_PRIVATE_KEY=
 
 HOME_GAS_PRICE_ORACLE_URL=https://gasprice.poa.network/
```

Comment:
Here is one of the comment we received during the security audit is: https://hackmd.io/s/Sk6P7Uqtm#2-Bad-private-data-storing. 
It is related to an environment when watcher and senders are run in the docker containers.
In order to address the comment I suggest the following:
1. For the environment when the watchers and senders are run on the same system (not containerized) `VALIDATOR_ADDRESS` is redundant - it can be calculated from the `VALIDATOR_ADDRESS_PRIVATE_KEY`. The official recommendation will be to put have `VALIDATOR_ADDRESS_PRIVATE_KEY` only in the configuration file. Moreover it will reduce the number of issues when a developer puts the validator address which does not correspond to the private key.
2. If the watcher finds that `VALIDATOR_ADDRESS_PRIVATE_KEY` is not configured (it means that it run in the container), it should use `VALIDATOR_ADDRESS`.

As soon as these changes are done, we will re-write the deployment playbooks as so they will store `VALIDATOR_ADDRESS_PRIVATE_KEY` in sender's containers only. Also the bridge startup script will calculate `VALIDATOR_ADDRESS` from `VALIDATOR_ADDRESS_PRIVATE_KEY` programmatically and initialize the watcher's contaners as so they will have `VALIDATOR_ADDRESS` without mentioning `VALIDATOR_ADDRESS_PRIVATE_KEY`.

What do you think?

Reply:
Let me see if I understood this correctly:

- Senders extract the `VALIDATOR_ADDRESS_PRIVATE_KEY` from the environment variables and use it to calculate the validator address.
- Watchers extract `VALIDATOR_ADDRESS` from the environment: this will be injected by the startup script. Also watchers should work fine if the private key is not present in the environment variables (since they don't use it anyway)

Did I get this right?

Reply:
Right.
And watchers should work in the case if `VALIDATOR_ADDRESS` is not specified (the case when watchers and senders are run in the same environment - not containerized) - `VALIDATOR_ADDRESS_PRIVATE_KEY` is used by watchers to calculate the validator address.

Reply:
I see, thanks!

Reply:
@fvictorio when could we expect these set of changes? I need  to send them for review to the security auditor to response on the opened finding.

Reply:
Done in 6575f0fd2cc8883f042bf2cd94d63840ad8031f0, but notice that the signature requests watcher still needs the private key:

https://github.com/poanetwork/token-bridge/blob/2fad9b9cca14b3eedbcecc8c931c845906c70bc4/src/events/processSignatureRequests/index.js#L60

Do the watchers **have** to work without the private key? Maybe there's some way to rewrite this so we don't need it.

Reply:
Ok. Let's leave the ability to work with the private key for the `signature-requests` watcher. Later we will think how to split this logic.